R/draws.R

```diff
@@ -1,10 +1,97 @@
+#' according to the given multiple imputation methodology and returns the parameter
+#' estimates from the fit.
 #'
 #' @name draws
-#' @param data TODO
-#' @param data_ice TODO
-#' @param vars TODO
-#' @param method TODO
+#' @param data A `data.frame` containing the data to be used in the model. See details.
+#' @param data_ice A `data.frame` that specifies the information related
+#' to the intercurrent events (ICE) and the imputation strategies. See details.
+#' @param vars A `vars` object as generated by [set_vars()]. See details.
+#' @param method A `method` object as generated by either [method_bayes()],
+#' [method_approxbayes()] or [method_condmean()].
+#' It specifies the multiple imputation methodology to be used. See details.
+#'
+#' @details
+#'
+#' `draws` performs the first step of the multiple imputation (MI) procedure: fitting the
+#' base imputation model. The goal is to estimate the parameters of interest needed
+#' for the imputation phase (i.e. the regression coefficients and the covariance matrices
+#' from a MMRM model).
+#'
+#' The function distinguishes between the following methods:
+#' - Bayesian MI based on MCMC sampling: `draws` returns the draws
+#' from the posterior distribution of the parameters using a Bayesian approach based on
+#' MCMC sampling. This method can be specified by using `method = method_bayes()`.
+#' - Approximate Bayesian MI based on bootstrap: `draws` returns
+#' the draws from the posterior distribution of the parameters using a Bayesian approach,
+#' where the sampling from the posterior distribution is simulated by fitting the MMRM model
+#' on bootstrap samples of the original dataset. This method can be specified by using `method = method_approxbayes()]`.
+#' - Conditional mean imputation with bootstrap re-sampling: `draws` returns the
+#' MMRM parameter estimates from the original dataset and from `n_samples` bootstrap samples.
+#' This method can be specified by using `method = method_condmean()` with
+#' argument `type = "bootstrap"`.
+#' - Conditional mean imputation with Jackknife re-sampling: `draws` returns the
+#' MMRM parameter estimates from the original dataset and from each leave-one-out sample.
+#' This method can be specified by using `method = method_condmean()` with
+#' argument `type = "jackknife"`.
+#'
+#' The argument `data` must have one row per visit per subject. This means that incomplete
+#' outcome data must be set as `NA` instead of having the related row missing. Missing values
+#' in the covariates are not allowed. If `data` is incomplete
+#' then the [expand_locf()] helper function can be used to add in any missing rows, using
+#' Last Observation Carried Forward (LOCF) imputation to impute the covariates values.
+#'
+#' The argument `data_ice` contains information about the occurrence of ICEs. It is a
+#' `data.frame` with 3 columns:
+#' - A vector of characters related to the ids of the subjects that experienced
```

Comment:
Same as my previous comment I would remove the word  "related" and state more clearly what these variables represent. Perhaps something like: 

The argument `data_ice` contains information about the occurrence of ICEs. It is a `data.frame` with 3 columns:
- Subject ID - a character vector containing the subject ID for which the corresponding ICE belongs to. This column must be named as specified in `vars$subjid`
- Visit - A character vector container the visit in which the ICE occoured at. This column must be named as specified in `vars$visit`
- Strategy - A chacater vector specifying which imputation strategy should be used for this subject. This column must be named as specified in `vars$strategy` possible imputation strategies are: ...

Reply:
Sure, with very little changes from your version I just addressed this.